pandas/core/dtypes/cast.py

```diff
@@ -159,6 +159,31 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+# TODO: this should be a TypeVar
+def maybe_box_native(value: Union[ABCSeries, Scalar]) -> Union[ABCSeries, Scalar]:
+    """
+    If passed a scalar cast the scalar to a python native type.
+
+    Parameters
+    ----------
+    value : scalar or Series
+
+    Returns
+    -------
+    scalar or Series
+    """
+    if not is_scalar(value):
+        pass
+    elif is_datetime_or_timedelta_dtype(value):
+        value = maybe_box_datetimelike(value)
+    elif is_float_dtype(value):
+        value = float(value)
+    elif is_integer_dtype(value):
```

Comment:
need to handle unsigned integers as well

Reply:
Added some test cases below and they pass as is. Are they what you meant?